pandas/core/arrays/categorical.py

```diff
@@ -2441,19 +2444,29 @@ def replace(self, to_replace, value, inplace: bool = False):
         """
         inplace = validate_bool_kwarg(inplace, "inplace")
         cat = self if inplace else self.copy()
-        if to_replace in cat.categories:
-            if isna(value):
-                cat.remove_categories(to_replace, inplace=True)
+        if is_list_like(to_replace):
+            if is_list_like(value):
+                if len(to_replace) != len(value):
+                    raise ValueError("to_replace and value must be same length!")
```

Comment:
Could this be a bit more precise? As in, include something about the lengths of `to_replace` and `value` which were received

Reply:
sorry, this and the change above it were experiments in addressing the dict case and list-list case for categoricals, which turned out to be more complex than expected and are not intended to be in this pr